Mathlib/Algebra/Group/Defs.lean

```diff
@@ -0,0 +1,136 @@
+/-!
+# Typeclasses for monoids and groups etc
+-/
+
+/-
+## Stuff which was in core Lean 3
+-/
+
+class Zero (α : Type u) where
+  zero : α
+
+instance [Zero α] : OfNat α (nat_lit 0) where
+  ofNat := Zero.zero
+
+class One (α : Type u) where
+  one : α
+
+instance [One α] : OfNat α (nat_lit 1) where
+  ofNat := One.one
+
+class Inv (α : Type u) where
+  inv : α → α
+
+postfix:max "⁻¹" => Inv.inv
+
+/-
+## The trick for adding a natural action onto monoids
+-/
+
+section nat_action
+
+variable {M : Type u}
+
+notation "ℕ" => Nat
```

Comment:
This notation is already declared in `Data.Nat.Basic`, import that instead

Reply:
But in mathlib3 data.nat.basic imports algebra.group.defs. Is this an issue?

Reply:
it will be, but we can sort out the dependencies later. Probably we will need some kind of prelude folder for really early stuff that should have been in core and is cheap to import everywhere, and this will be in it.

Reply:
I proved the basic lemmas for `Group` but I need to go back to real life stuff (marking exams) so I commented out the sorrys. I can come back to this next Thursday (my Lean 4 day). In the absence of to_additive I'll have to manually port the Group API (about 10 lemmas) to AddGroup and then prove zero_mul and mul_zero for rings and then CommRing will be easy and I'll have finished Sebastien's sketch of what the hierarchy should look like up to CommRing.

Reply:
OK I fixed the import. I still define \Z for Int though. I agree with the prelude idea.